src/components/Snippet/index.tsx

```diff
@@ -0,0 +1,240 @@
+      : ''};
+`
+
+const ShowMoreContainer = styled('div', {
+  shouldForwardProp: prop => !['showMore'].includes(prop),
+})<{ showMore?: boolean }>`
+  width: 100%;
+  box-shadow: ${({ theme, showMore }) =>
+    !showMore
+      ? `0px -22px 19px -6px
+    ${theme.colors.neutral.backgroundStrong}`
+      : 'none'};
+`
+
+const StyledButton = styled.button`
+  width: 100%;
+  background: none;
+  border: none;
+  padding: ${({ theme }) => theme.space['2']};
+  padding-top: ${({ theme }) => theme.space['1']};
+  cursor: pointer;
+`
+
+const AlignCenterText = styled(Text)`
+  display: flex;
+  justify-content: center;
+  align-items: center;
+`
+
+const AnimatedArrowIcon = styled(Icon, {
+  shouldForwardProp: prop => !['showMore'].includes(prop),
+})<{ showMore?: boolean }>`
+  transform: ${({ showMore }) =>
+    showMore ? 'rotate(180deg)' : 'rotate(0deg)'};
+  transform-origin: center;
+  transition: transform 300ms ease-in-out;
+`
+
+type SnippetProps = {
+  value: string
+  multiline?: boolean
+  /**
+   * prefix display an element at the beginning of the snippet that is not copiable or selectable.
+   * For `lines` prefix it will display the line number.
+   * For `command` prefix it will display a `$` sign.
+   */
+  prefix?: Prefixes
+  showText?: string
+  hideText?: string
+} & Pick<CopyButtonProps, 'copyText' | 'copiedText'>
+
+const Snippet = ({
+  value,
+  copyText,
+  copiedText,
+  multiline = false,
+  showText = 'Show',
+  hideText = 'Hide',
+  prefix,
+}: SnippetProps) => {
+  const theme = useTheme()
+  const [showMore, setShowMore] = useState(false)
+
+  const lines = value.split(LINES_BREAK_REGEX).filter(Boolean)
+  const numberOfLines = lines.length
+  const hasShowMoreButton = numberOfLines > 4 && multiline
+
+  /**
+   * This is used to calculate the max height of the snippet in order to add an animation when the show more button is clicked.
+   * The max height is calculated by multiplying the number of lines by the line height of the text plus the padding (2 times).
+   */
+  const maxHeightExtended =
+    numberOfLines * Number(theme.typography.code.lineHeight.replace('px', '')) +
+    Number(theme.space['2'].replace('px', '')) * 2
+
+  return (
+    <Container multiline={multiline}>
+      <StyledStack>
+        <PreText
+          as="pre"
+          variant="code"
+          multiline={multiline}
+          hasShowMoreButton={hasShowMoreButton}
+          showMore={showMore}
+          maxHeightExtended={maxHeightExtended}
+        >
+          {multiline ? (
+            lines.map((line, index) => (
+              <StyledSpan multiline={multiline} key={line}>
+                {prefix ? (
+                  <Prefix
+                    as="span"
+                    variant="code"
+                    prefix={prefix}
+                    prominence="weak"
+                  >
+                    {prefix === 'lines' ? `${index + 1}` : '$'}
+                  </Prefix>
```

Comment:
Might be better to use css counters :)
Meaning user will be able to select text by selection it

https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Counter_Styles/Using_CSS_counters

Reply:
@chambo-e I changed it, now we put those values into a pseudo element `:before` with CSS counter or `$` text depending on `prefix` property. It also make the JSX smaller and I cloud remove a styled const 🎉